src/google-maps/map-info-window/map-info-window.ts

```diff
@@ -0,0 +1,184 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+  ViewChild
+} from '@angular/core';
+import {BehaviorSubject, combineLatest, Observable, ReplaySubject, Subject} from 'rxjs';
+import {map, takeUntil} from 'rxjs/operators';
+
+import {GoogleMap} from '../google-map/index';
+import {MapMarker} from '../map-marker/index';
+
+/**
+ * Angular component that renders a Google Maps info window via the Google Maps JavaScript API.
+ * @see developers.google.com/maps/documentation/javascript/reference/info-window
+ */
+@Component({
+  selector: 'map-info-window',
+  template: `<div class="map-info-window-container">
```

Comment:
I'm not sure any of this template is necessary; I tried modifying this to use only the host element and it seems to work:
https://github.com/jelbourn/components/commit/415738c7e9c625992b033db273fd05ff5a41510c

Is there something I'm missing that the additional element adds?

Reply:
Discussed in the other comment, it allows for a more specific selector so that display: none prevents display in the dom, but not in the map.

Reply:
Okay, now that I actually understand what this was doing, here's my proposed change to use a single element:
https://github.com/jelbourn/components/commit/c66418e841bf3ca59bed060877aa59e1f7eb6a4f

This has the benefits:
* Fewer elements for Angular to render / generate code for
* Doesn't leave hanging empty elements inside the `<google-map>` element
* Classes and attributes applied to `<map-info-window>` will apply
* Can use `@Directive` instead of `@Component`, which generates less code and does less run-time work
* Doing `querySelector` for `map-info-window` will show up where you'd expect it to (in the map) and contain the expected content

Admittedly the it's all pretty minor, but I think it's worth it. `<mat-menu>` currently has this problem of being a "fake" element whose content gets hoisted off elsewhere, leaving an empty element behind that sometimes confuses people.

Reply:
Good idea. Done